### PR TITLE
Migrate Pin authorization flow to Jetpack Compose

### PR DESCRIPTION
**Pull request by prudrabhat**
## Description

Assurance extension UI is currently built using XML based design system using webviews. As part of the UI modernization effort, it will be migrated to Jetpack Composed based UI. This PR contains changes that switch Assurance PIN authorization flow to Compose UI and the floating button UI to MobileCore's v-next UI Service that uses Compose.

#### Key Concepts / Changes:
- Currently pin authorization creates the `AssuranceSession` first and presents the containing PIN UI. QuickConnect authorization flow presents the quick-connect UI first and then creates the `AssuranceSession`. This divergent control flow between Pin and QuickConnect authorization is now unified and is now UI driven. More specifically the authorization UI is always presented first and the `AssuranceSession` is only created when authorization and connection is complete. This allows AssuranceSession to own the status ui (floating button and status screen) and separates the authorizatio ui. The control flow for the session creation now is : `startSession()` api parses details associated with the authorization --> appropriate screen for the authorization is launched --> authorization screen action will determine the creation or termination of the assurance session.
- UI that the SDK displays is governed by the `AssuranceAppState`. More specifically the `AssuranceAppState.SessionPhase`. `SessionPhase` represents the state of the AssuranceSession as understood by the SDK. The `SessionPhase` starts from `Disconnected`  which is a state where the session is not yet established or the session has been disconnected (either normally or due to an error). It transitions to `Authorizing` when a connection is initiated (via the `startSession()` api). A successful socket connection transitions the state to `Connected` and a failed connection transitions back to  `Disconnected` or `Authorizing` in case of a retry possibility.
- While `SessionPhase` is used to launch `AssuranceActivity` with the  right *screen*, `AssuranceSessionStatusListener` is now the default way for the UI to get notified of the connection status of the session. Its use for managing the outbound buffer is retained.
- `AssuranceActivity` displays all Assurance SDK screens. `AssuranceFullScreenTakeoverActivity` and `AssuranceErrorDisplayActivity` are now removed. They are replaced by `PinScreen` , `AssuranceErrorScreen`, `AssuranceStatusScreen` composables. This implies that Assurance  no longer uses html to display UI. The PR following up this will also remove `AssuranceQuickConnectActivity` in favor of `QuickConnectScreen`.
- `AssuranceNavHost` controls the navigation to the right screen via `AssuranceDestination` path resulting from the session phase.
- `AssuranceComponentRegistry` holds the `SessionUIOperationHandler` which acts a glue between the presentation layer and the AssuranceSession. `SessionUIOperationHandler` has been refactored to remove individual methods for pin and quick connect now that the flow is unified.
- The current implementation of the floating-button that depicts the status of Assurance session is now replaced with the Compose based implementation utility from Core via `UIService`
- `AssuranceSessionPresentationManager` is responsible for managing the floating button, status UI and the `SessionPhase` starting from when the session is created. It is sufficient to replace the `AssuranceSessionAtuthorizingPresentation` which is now removed.Note that the loggin operations for session status will be implemented in a followup PR.
- Threading details for interactions between AssuranceSession and UI will be ironed out when all of the UI is in Compose.

#### UI Component Changes
- Assurance UI is now made of 4 screens 
  - `PinScreen` - Landing screen for the Pin authorization flow. It displays dialpad and Connect and Cancel action buttons. It has a `PinScreenViewModel` that acts on the various `PinScreenAction`'s from the ui. 
  - `AssuranceErrorScreen` - An error screen that is displayed when a non retryable error occurs while the authorization ui (like the `PinScreen` or `QuickConnect` screen) are not active.
  - `AssuranceStatusScreen` - The screen that displays informational logs about the session connectivity and plugin action.
  - `QuickConnectScreen` - To be implemented. Landing screen for the quick connect flow.

- `PinScreen` is made of `DialPadView`, `PinErrorView`, `PinConnecting` views to be displayed during the session connection trasitions. `PinScreenViewModel` and `PinScreenState` influence the view being shown.
- `AssuranceFloatingButton` is a wrapper around the floating button provided bu `UIService` with a couple methods to change the type of graphic being displayed.

#### Other Changes

- `AssuranceConstants` , `AssuranceStateManager`, `AssuranceSharedStateManager` have now been migrated to Kotlin. The contents will be improved during the course of the refactor.
- `AssuranceSharedState` represents the shared state of Assurance.
- `ApplicationHandle` which was a utlity class for accessing Application and Activity has been deleted infavor of `ServiceProvider.AppContextService` from Core.

**Note:**
- Units tests + instrumentation tests modified and new tests added as a follow up.
- Status UI logging will be implemented as follow up.

## Related Issue
#78 

## Motivation and Context
Assurance extension UI is currently built using XML based design system using webviews. As part of the UI modernization effort, it needs to be migrated to Jetpack Composed based UI. This issue tracks changes to switch Assurance PIN authorization flow to Compose UI and the floating button UI to MobileCore's v-next UI Service that uses Compose.

## How Has This Been Tested?
- Manual tests
Unit and instrumentation tests will be added in a separate PR.

## Screenshots (if appropriate):
| Pin Screen Half Filled | Pin Screen Filled | Pin Screen Error |
|------------------------|-------------------|-------------------|
|  |  |  |

| Loading Screen | Connected Screen | Status Screen |
|-----------------|------------------|---------------|
|  |  |  |
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by praveek** [comment]
I am not sure if you have planned this for future PR but I suggest consolidating the UI initiation logic within `AssuranceSessionOrchestrator`. Currently, this logic is scattered among AssuranceExtension, AssuranceSessionPresentationManager, and AssuranceSessionOrchestrator. Centralizing it in AssuranceSessionOrchestrator will make it easier to follow the flow.

**Comment by prudrabhat** [comment]
> I am not sure if you have planned this for future PR but I suggest consolidating the UI initiation logic within `AssuranceSessionOrchestrator`. Currently, this logic is scattered among AssuranceExtension, AssuranceSessionPresentationManager, and AssuranceSessionOrchestrator. Centralizing it in AssuranceSessionOrchestrator will make it easier to maintain.

I intended to move the UI launch triggers for  (PIN, QuickConnect) to a single place inside the `AssuranceExtension` as they are triggered due to events and leave the `AssuranceFloatingButton` management in the session presentation manager due to these locations being the sources of initiation. But, lets discuss about how moving it to AssuranceSessionOrchestrator will simplify things and then I can do it as part of the next PR based on the outcome.

**Review by praveek — APPROVED** [review]

**Review by praveek — APPROVED** [review]

**Review by praveek — APPROVED** [review]

File: code/assurance/src/main/java/com/adobe/marketing/mobile/assurance/AssuranceSharedStateManager.kt [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            Log.warning(
+                Assurance.LOG_TAG,
+                LOG_TAG,
+                "Assurance datastore is null, unable to persist assurance state."
+            )
+            return
+        }
+
+        if (stateToPersist.clientId.isBlank()) {
+            assuranceDataStore.remove(AssuranceConstants.DataStoreKeys.CLIENT_ID)
```
[the remark this answers is not in the document]
Comment (prudrabhat) [reply code/assurance/src/main/java/com/adobe/marketing/mobile/assurance/AssuranceSharedStateManager.kt:105]: You are correct, this is just for completeness. We currently do not allow clearing the clientId.

File: code/assurance/src/main/java/com/adobe/marketing/mobile/assurance/AssuranceSharedStateManager.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    /**
+     * Loads the Assurance shared state from persistence if one was previously stored.
+     * @return the Assurance shared state if one was previously stored, otherwise a creates a
+     * new empty [AssuranceSharedState]
+     */
+    private fun loadFromPersistence(): AssuranceSharedState {
+        if (assuranceDataStore == null) {
+            return AssuranceSharedState("", "")
```
[the remark this answers is not in the document]
Comment (prudrabhat) [reply code/assurance/src/main/java/com/adobe/marketing/mobile/assurance/AssuranceSharedStateManager.kt]: Changed this to named parameters.

**Pull request merged**